src/nodes.coffee

```diff
@@ -1203,6 +1185,33 @@ exports.Value = class Value extends Base
     else
       @error 'tried to assign to unassignable value'
 
+  astType: ->
+    unless @hasProperties()
+      @base.astType()
+    else
+      'MemberExpression'
+
+  astProperties: ->
+    unless @hasProperties()
+      return @base.astProperties()
+    else
+      # If this `Value` has properties, the *last* property (e.g. `c` in `a.b.c`)
+      # becomes the `property`, and the preceding properties (e.g. `a.b`) become
+      # a child `Value` node assigned to the `object` property.
+      property = @properties[@properties.length - 1]
+      return
+        object: new Value(@base.unwrap(), @properties[0...(@properties.length - 1)], @tag, @isDefaultValue).ast()
+        property: property.ast()
+        computed: property instanceof Index or property.name?.unwrap() not instanceof PropertyName
+        optional: !!property.soak
+        shorthand: !!property.shorthand
+
+  astLocationData: ->
+    unless @hasProperties()
```

Comment:
> I ended up with something pretty close to what you were describing in your comments

Right, that was a warning, not a recommendation. By insisting on not just overriding `ast()` you're now suffering from unnecessary "parallelism" - you have to duplicate
```
unless @hasProperties()
  @base.whatever()
else
  ...
```
in three places. In zero ways do I find that elegant or desirable

It's obviously harder to read and understand the shape of what's actually getting produced (which is the goal here with the `ast` methods)